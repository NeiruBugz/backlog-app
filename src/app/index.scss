@import './styles/index.scss';

:root {
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu,
    Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 14px;
  line-height: 24px;
  font-weight: 400;

  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
}

*,
*::after,
*::before {
  box-sizing: border-box;
  scroll-behavior: smooth;
}

#root {
  height: 100vh;
  width: 100%;
}

body {
  margin: 0;
  place-items: center;
  min-width: 320px;
  min-height: 100vh;
}

.card.image-full:before {
  background-color: transparent !important;
}
