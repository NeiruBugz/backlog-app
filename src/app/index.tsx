import { withProviders } from './providers';
import { Routing } from '@pages';

import './index.scss';

const App = () => <Routing />;

export default withProviders(App);
