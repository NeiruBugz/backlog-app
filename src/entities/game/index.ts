export * from './types';
export * from './ui/card';
export * from './ui/card-list';
