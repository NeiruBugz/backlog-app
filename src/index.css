:root {
  font-family: Inter, Avenir, Helvetica, Arial, sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  color: rgba(255, 255, 255, 0.87);
  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
  --ba-modal-index: 1000;
}

a, button {
  cursor: pointer;
}

.container {
  padding: 16px 20px 0;
}

.ba-button {
  border: 1px solid #242424;
  border-radius: 4px;
  background: transparent;
  text-align: center;
  padding: 4px 8px;
  transition: all .6ms ease-in-out;
}

.ba-button--primary {
  background-color: #242424;
  color: #fdfdfd;
}

.ba-button--primary:hover {
  background-color: transparent;
  color: #242424;
}

.ba-button--secondary:hover {
  border-color: transparent;
}

.ba-modal-overlay {
  width: 100vw;
  height: 100vh;
  position: absolute;
  z-index: var(--ba-modal-index, 1000);
  top: 0;
  left: 0;
  background-color: #24242442;
  display: flex;
  justify-content: center;
  align-items: center;
}

.ba-modal {
  background-color: #fdfdfd;
  display: flex;
  flex-direction: column;
  color: #242424;
}

.ba-modal-header {
  text-align: center;
  font-weight: bold;
  font-size: 20px;
  padding: 12px 24px 0;
}

.ba-modal-body {
  padding: 0 24px 12px;
}

.ba-label {
  display: block;
}

.ba-input {
  display: block;
}