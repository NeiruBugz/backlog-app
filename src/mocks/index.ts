import type { Game } from '@entities';

const cardProps: Game = {
  title: 'Persona 5 Royal',
  img: 'https://assets1.ignimgs.com/2020/02/14/persona-5-royal---button-fin-1581716582492.jpg',
  platform: 'Nintendo Switch',
  status: 'in-progress',
};

const gamesList: Game[] = [
  {
    title: 'God of War: Ragnar√∂k',
    img: '',
    platform: 'PlayStation 5',
    status: 'completed',
  },
  {
    title: 'The Last Of Us Part I',
    img: '',
    platform: 'PlayStation 5',
    status: 'completed',
  },
  {
    title: 'Forza Horizon 5',
    img: '',
    platform: 'Xbox',
    status: 'backlog',
  },
  cardProps,
];

const user = {
  username: 'NeiruBugz',
  avatarUrl: '',
};

export { cardProps, gamesList, user };
