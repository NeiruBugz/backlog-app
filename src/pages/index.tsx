import { Navigate, Route, Routes } from 'react-router';
import { useStore } from '@nanostores/react';
import { Home } from './home';
import { Auth } from './auth';
import { AddGame } from './add-game';
import { GamesList } from './games-list';
import { Header } from '@widgets';
import { nanoUser } from 'entities/user/slice/index';

const ProtectedRoute = ({ children }: { children: JSX.Element }) => {
  const { authorized } = useStore(nanoUser);

  if (!authorized) {
    return <Navigate to="/" replace />;
  }

  return children;
};

const Routing = () => (
  <div className="container mx-auto">
    <Header />
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/auth" element={<Auth />} />
      <Route
        path="/list"
        element={
          <ProtectedRoute>
            <GamesList />
          </ProtectedRoute>
        }
      />
      <Route
        path="/add-game"
        element={
          <ProtectedRoute>
            <AddGame />
          </ProtectedRoute>
        }
      />
      <Route path="*" element={<p>There&apos;s nothing here: 404!</p>} />
    </Routes>
  </div>
);
export { Routing };
