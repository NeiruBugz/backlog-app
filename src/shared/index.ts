export * from './api';
export * from './types';
export * from './helpers';
export * from './store';
export * from './hooks';
export * from './api/firebase';
