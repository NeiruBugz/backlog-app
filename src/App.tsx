import 'normalize.css/normalize.css';
import './App.css';
import { useCallback, useState } from 'react';
import { Layout, Button, Space } from 'antd';

import { Card, AuthModal, Header } from '@components';
import { useEscapeKeyPress } from './hooks/useEscapeKeyPress';
import { useList } from 'effector-react';
import { addGame, gamesList } from '@store';
import { cardProps } from './mocks/data/card.mock';
import { Game } from './types/data/game';

const { Content, Sider } = Layout;

function App() {
  const [modalOpen, setModalOpen] = useState(false);
  useEscapeKeyPress(useCallback(() => setModalOpen(false), []));

  return (
    <>
      <Layout style={{ height: '100vh' }}>
        <Header setModalOpen={setModalOpen} />
        <Layout>
          <Sider width={200}>Sider</Sider>
          <Content>
            <div className="container">
              <Button type='primary' onClick={() => addGame(cardProps as Game)}>Add Persona 5</Button>
              <Space wrap>
                {useList(gamesList, (game) => <Card {...game} />)}
              </Space>
            </div>
          </Content>
        </Layout>
      </Layout>
      <AuthModal isOpen={modalOpen} onCancel={() => setModalOpen(false)} />
    </>
  );
}

export default App;
