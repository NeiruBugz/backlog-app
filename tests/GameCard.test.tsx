import { GameCard } from '../src/entities';
import { render } from '@testing-library/react';
import { cardProps } from '../src/mocks';

describe('GameCard', () => {
  it('snapshots', () => {
    const { container } = render(<GameCard {...cardProps}/>);
    expect(container).toMatchSnapshot();
  });
});