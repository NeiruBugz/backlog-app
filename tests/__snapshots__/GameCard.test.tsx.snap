// Vitest Snapshot v1

exports[`GameCard > snapshots 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered _ba-card_53d2b9 css-dev-only-do-not-override-sk7ap8"
  >
    <div
      class="ant-card-body"
    >
      <h4>
        Persona 5 Royal
      </h4>
      <div>
        <span
          class="ant-tag _ba-card__platform_53d2b9 _ba-card__platform--nintendo_53d2b9 css-dev-only-do-not-override-sk7ap8"
        >
          Nintendo Switch
        </span>
        <button
          class="ant-btn css-dev-only-do-not-override-sk7ap8 ant-btn-default ant-dropdown-trigger"
          type="button"
        >
          <div
            class="ant-space css-dev-only-do-not-override-sk7ap8 ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 8px;"
            >
              Move to
            </div>
            <div
              class="ant-space-item"
            >
              <span
                aria-label="down"
                class="anticon anticon-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
